File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.py
class Solution:
    def isPalli(self, word: str) -> bool:
        newWord = word[::-1]
        return word == newWord

    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            if self.isPalli(word):
                return word
        return ""

File: LeetCode/2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for (string word : words) {
            int n = word.size();
            if (equal(word.begin(), word.begin() + n/2, word.rbegin())) {
                return word;
            }
        }
        return "";
    }
};

// class Solution {
// public:
//     bool isPalli(string word) {
//         string newWord = word;
//         reverse(newWord.begin(), newWord.end());
//         return word == newWord;
//     }

//     string firstPalindrome(vector<string>& words) {
//         for (string word : words) {
//             if (isPalli(word)) {
//                 return word;
//             }
//         }
//         return "";
//     }
// };